// @notionhq/client/build/src/api-endpoints/comments.js
"use strict";
// cspell:disable-file
// Note: This is a generated file. DO NOT EDIT!
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteComment = exports.updateComment = exports.getComment = exports.listComments = exports.createComment = void 0;
/**
 * Create a comment
 */
exports.createComment = {
    method: "post",
    pathParams: [],
    queryParams: [],
    bodyParams: [
        "attachments",
        "display_name",
        "parent",
        "rich_text",
        "markdown",
        "discussion_id",
    ],
    path: () => `comments`,
};
/**
 * List comments
 */
exports.listComments = {
    method: "get",
    pathParams: [],
    queryParams: ["block_id", "start_cursor", "page_size"],
    bodyParams: [],
    path: () => `comments`,
};
/**
 * Retrieve a comment
 */
exports.getComment = {
    method: "get",
    pathParams: ["comment_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `comments/${p.comment_id}`,
};
/**
 * Update a comment
 */
exports.updateComment = {
    method: "patch",
    pathParams: ["comment_id"],
    queryParams: [],
    bodyParams: ["rich_text", "markdown"],
    path: (p) => `comments/${p.comment_id}`,
};
/**
 * Delete a comment
 */
exports.deleteComment = {
    method: "delete",
    pathParams: ["comment_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `comments/${p.comment_id}`,
};
