// @notionhq/client/build/src/api-endpoints/databases.js
"use strict";
// cspell:disable-file
// Note: This is a generated file. DO NOT EDIT!
Object.defineProperty(exports, "__esModule", { value: true });
exports.createDatabase = exports.updateDatabase = exports.getDatabase = void 0;
/**
 * Retrieve a database
 */
exports.getDatabase = {
    method: "get",
    pathParams: ["database_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `databases/${p.database_id}`,
};
/**
 * Update a database
 */
exports.updateDatabase = {
    method: "patch",
    pathParams: ["database_id"],
    queryParams: [],
    bodyParams: [
        "parent",
        "title",
        "description",
        "is_inline",
        "icon",
        "cover",
        "in_trash",
        "is_locked",
    ],
    path: (p) => `databases/${p.database_id}`,
};
/**
 * Create a database
 */
exports.createDatabase = {
    method: "post",
    pathParams: [],
    queryParams: [],
    bodyParams: [
        "parent",
        "title",
        "description",
        "is_inline",
        "initial_data_source",
        "icon",
        "cover",
    ],
    path: () => `databases`,
};
