// @notionhq/client/build/src/api-endpoints/data-sources.js
"use strict";
// cspell:disable-file
// Note: This is a generated file. DO NOT EDIT!
Object.defineProperty(exports, "__esModule", { value: true });
exports.listDataSourceTemplates = exports.createDataSource = exports.queryDataSource = exports.updateDataSource = exports.getDataSource = void 0;
/**
 * Retrieve a data source
 */
exports.getDataSource = {
    method: "get",
    pathParams: ["data_source_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `data_sources/${p.data_source_id}`,
};
/**
 * Update a data source
 */
exports.updateDataSource = {
    method: "patch",
    pathParams: ["data_source_id"],
    queryParams: [],
    bodyParams: ["archived", "title", "icon", "properties", "in_trash", "parent"],
    path: (p) => `data_sources/${p.data_source_id}`,
};
/**
 * Query a data source
 */
exports.queryDataSource = {
    method: "post",
    pathParams: ["data_source_id"],
    queryParams: ["filter_properties"],
    bodyParams: [
        "archived",
        "sorts",
        "filter",
        "start_cursor",
        "page_size",
        "in_trash",
        "result_type",
    ],
    path: (p) => `data_sources/${p.data_source_id}/query`,
};
/**
 * Create a data source
 */
exports.createDataSource = {
    method: "post",
    pathParams: [],
    queryParams: [],
    bodyParams: ["parent", "properties", "title", "icon"],
    path: () => `data_sources`,
};
/**
 * List templates in a data source
 */
exports.listDataSourceTemplates = {
    method: "get",
    pathParams: ["data_source_id"],
    queryParams: ["name", "start_cursor", "page_size"],
    bodyParams: [],
    path: (p) => `data_sources/${p.data_source_id}/templates`,
};
