// @notionhq/client/build/src/api-endpoints/file-uploads.js
"use strict";
// cspell:disable-file
// Note: This is a generated file. DO NOT EDIT!
Object.defineProperty(exports, "__esModule", { value: true });
exports.getFileUpload = exports.completeFileUpload = exports.sendFileUpload = exports.listFileUploads = exports.createFileUpload = void 0;
/**
 * Create a file upload
 */
exports.createFileUpload = {
    method: "post",
    pathParams: [],
    queryParams: [],
    bodyParams: [
        "mode",
        "filename",
        "content_type",
        "number_of_parts",
        "external_url",
    ],
    path: () => `file_uploads`,
};
/**
 * List file uploads
 */
exports.listFileUploads = {
    method: "get",
    pathParams: [],
    queryParams: ["status", "start_cursor", "page_size"],
    bodyParams: [],
    path: () => `file_uploads`,
};
/**
 * Upload a file
 */
exports.sendFileUpload = {
    method: "post",
    pathParams: ["file_upload_id"],
    queryParams: [],
    bodyParams: [],
    formDataParams: ["file", "part_number"],
    path: (p) => `file_uploads/${p.file_upload_id}/send`,
};
/**
 * Complete a multi-part file upload
 */
exports.completeFileUpload = {
    method: "post",
    pathParams: ["file_upload_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `file_uploads/${p.file_upload_id}/complete`,
};
/**
 * Retrieve a file upload
 */
exports.getFileUpload = {
    method: "get",
    pathParams: ["file_upload_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `file_uploads/${p.file_upload_id}`,
};
