// @notionhq/client/build/src/api-endpoints/views.js
"use strict";
// cspell:disable-file
// Note: This is a generated file. DO NOT EDIT!
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteViewQuery = exports.getViewQueryResults = exports.createViewQuery = exports.deleteView = exports.updateView = exports.getView = exports.createView = exports.listDatabaseViews = void 0;
/**
 * List views
 */
exports.listDatabaseViews = {
    method: "get",
    pathParams: [],
    queryParams: ["database_id", "data_source_id", "start_cursor", "page_size"],
    bodyParams: [],
    path: () => `views`,
};
/**
 * Create a view
 */
exports.createView = {
    method: "post",
    pathParams: [],
    queryParams: [],
    bodyParams: [],
    path: () => `views`,
};
/**
 * Retrieve a view
 */
exports.getView = {
    method: "get",
    pathParams: ["view_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `views/${p.view_id}`,
};
/**
 * Update a view
 */
exports.updateView = {
    method: "patch",
    pathParams: ["view_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `views/${p.view_id}`,
};
/**
 * Delete a view
 */
exports.deleteView = {
    method: "delete",
    pathParams: ["view_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `views/${p.view_id}`,
};
/**
 * Create a view query
 */
exports.createViewQuery = {
    method: "post",
    pathParams: ["view_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `views/${p.view_id}/queries`,
};
/**
 * Get view query results
 */
exports.getViewQueryResults = {
    method: "get",
    pathParams: ["view_id", "query_id"],
    queryParams: ["start_cursor", "page_size"],
    bodyParams: [],
    path: (p) => `views/${p.view_id}/queries/${p.query_id}`,
};
/**
 * Delete a view query
 */
exports.deleteViewQuery = {
    method: "delete",
    pathParams: ["view_id", "query_id"],
    queryParams: [],
    bodyParams: [],
    path: (p) => `views/${p.view_id}/queries/${p.query_id}`,
};
