// @notionhq/client/build/src/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MIN_VIEW_COLUMN_WIDTH = exports.DEFAULT_MAX_RETRY_DELAY_MS = exports.DEFAULT_INITIAL_RETRY_DELAY_MS = exports.DEFAULT_MAX_RETRIES = exports.DEFAULT_TIMEOUT_MS = exports.DEFAULT_BASE_URL = void 0;
/**
 * Default base URL for the Notion API.
 */
exports.DEFAULT_BASE_URL = "https://api.notion.com";
/**
 * Default request timeout in milliseconds (60 seconds).
 */
exports.DEFAULT_TIMEOUT_MS = 60000;
/**
 * Default maximum number of retry attempts for failed requests.
 */
exports.DEFAULT_MAX_RETRIES = 2;
/**
 * Default initial delay between retries in milliseconds (1 second).
 * Used as the base for exponential back-off when the retry-after
 * header is absent.
 */
exports.DEFAULT_INITIAL_RETRY_DELAY_MS = 1000;
/**
 * Default maximum delay between retries in milliseconds
 * (60 seconds). Caps both retry-after and exponential
 * back-off delays.
 */
exports.DEFAULT_MAX_RETRY_DELAY_MS = 60000;
/**
 * The minimum width of a view column in pixels. Use this
 * with the views API to make a property column that appears
 * minimal / collapsed in the Notion app UI (e.g. a checkbox
 * or status-as-checkbox column).
 */
exports.MIN_VIEW_COLUMN_WIDTH = 32;
